src/pages-and-resources/data/thunks.js

```diff
@@ -1,33 +1,51 @@
+import { RequestStatus } from '../../data/constants';
 import {
-  getPages,
+  getCourseApps,
+  updateAppStatus,
 } from './api';
-import { addModels } from '../../generic/model-store';
+import { addModels, updateModel } from '../../generic/model-store';
 import {
-  FAILED,
-  fetchPagesSuccess,
-  LOADING,
-  updateStatus,
-  LOADED,
+  fetchCourseAppsSuccess,
+  updateLoadingStatus,
+  updateSavingStatus,
 } from './slice';
 
 /* eslint-disable import/prefer-default-export */
-export function fetchPages(courseId) {
+export function fetchCourseApps(courseId) {
   return async (dispatch) => {
-    dispatch(updateStatus({ courseId, status: LOADING }));
+    dispatch(updateLoadingStatus({ courseId, status: RequestStatus.IN_PROGRESS }));
 
     try {
-      const { pages } = await getPages(courseId);
+      const courseApps = await getCourseApps(courseId);
 
-      dispatch(addModels({ modelType: 'pages', models: pages }));
-      dispatch(fetchPagesSuccess({
-        pageIds: pages.map(page => page.id),
+      dispatch(addModels({ modelType: 'courseApps', models: courseApps }));
+      dispatch(fetchCourseAppsSuccess({
+        courseAppIds: courseApps.map(courseApp => courseApp.id),
       }));
-      dispatch(updateStatus({ courseId, status: LOADED }));
+      dispatch(updateLoadingStatus({ courseId, status: RequestStatus.SUCCESSFUL }));
     } catch (error) {
       // TODO: We need generic error handling in the app for when a request just fails... in other
       // parts of the app (proctored exam settings) we show a nice message and ask the user to
       // reload/try again later.
-      dispatch(updateStatus({ courseId, status: FAILED }));
+      dispatch(updateLoadingStatus({ courseId, status: RequestStatus.FAILED }));
+    }
+  };
+}
+
+export function enableDisableApp(courseId, appId, state, syncChange = false) {
+  return async (dispatch) => {
+    if (!syncChange) {
```

Comment:
Why this piece of `if` condition needed here? When we only need to update store and not make an API call? Can you elaborate?

Reply:
Yes, it's used to persist state to the store without changing the state on the server.

Reply:
Whats' the use case for this?

Reply:
Hmm, I think it was used in a previous version of the code that got refactored out. 

But generally the idea was, that if a user is  applying complex changes to something like teams, you keep all those changes in the store, and save then so if the page accidentally closes, you don't lose those changes. You can then pick up where you left off and sync.